src/builder/module_builder.rs

```diff
@@ -72,34 +73,33 @@ impl<T: HugrMutRef> ModuleBuilder<T> {
         f_id: &FuncID<false>,
     ) -> Result<FunctionBuilder<&mut HugrMut>, BuildError> {
         let f_node = f_id.node();
-        let (inputs, outputs) = if let OpType::Module(ModuleOp::Declare { signature }) =
-            self.hugr().get_optype(f_node)
-        {
-            (signature.input.clone(), signature.output.clone())
-        } else {
-            return Err(BuildError::UnexpectedType {
-                node: f_node,
-                op_desc: "ModuleOp::Declare",
-            });
-        };
+        let (inputs, outputs) =
+            if let OpType::Declare(ops::Declare { signature }) = self.hugr().get_optype(f_node) {
+                (signature.input.clone(), signature.output.clone())
+            } else {
+                return Err(BuildError::UnexpectedType {
+                    node: f_node,
+                    op_desc: "OpType::Declare",
+                });
+            };
```

Comment:
Could be done by checking
`OpTag::Function.contains(optype.tag())`
(I'm not sure it it would make things simpler here)